### PR TITLE
feat: CICD Automate Generation and Deployment of Full St (#29160)

### LINKED ISSUE #29160: CICD: Automate starter generation
### Parent Issue

_No response_

### Task

Generating our full and empty starters has always been a manual process. We must identify what part of the process can be automated with GH actions. It will make the process more efficient and easy for everyone. Take into consideration that we have an endpoint to generate a starter file from a dotCMS instance (it might help to get a full starter, but it won't work for empty starters). It might not be possible to trigger the generation of both starters from GH, but at least we should automate the publication on Artifactory and a Slack notification

Ideally, the action should: 
- Ask for a summary of changes included on the starter to be published
- Type of starter: empty, full? 
- Download the starter and publish it on Artifactory
- Notify on Slack

### Proposed Objective

Code Maintenance

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

_No response_

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by dcolina**
### Proposed Changes
* This GitHub Actions workflow automates the process of publishing a starter package. It is triggered manually and includes three main jobs:

- Get Full Starter: Downloads the starter package from a URL and uploads it as an artifact.
- Deploy Artifacts: Deploys the downloaded artifact to Artifactory, handling dry-run mode if specified.
- Send Notification: Sends a Slack notification about the newly published starter package if the deployment was successful and dry-run mode is disabled.

*Note*: Initially, runners are whitelisted to access the Starter server via a script managed by the Cloud Engineering team. 

### Required secrets and variables

*Secrets*:

- `DOT_STARTER_ACCESS_TOKEN` into the *trunk* environment. It must be a valid API Token (from the Starter server).
- `ARTIFACTORY_ACCESS_TOKEN` into the *trunk* environment.

*Variables*:

- `DOT_STARTER_URL` into the *trunk* environment.
- `ARTIFACTORY_REPO` into the *trunk* environment. It must be `libs-release-local`
- `ARTIFACTORY_URL` into the *trunk* environment. It must be `https://repo.dotcms.com`

### Additional Info
Related to #29160 (CICD Automate Generation and Deployment of Full Starter).

This PR fixes: #29160

### CODE REVIEW

File: .github/workflows/publish-starter.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+jobs:
+  get-full-starter:
+    runs-on: macos-13
+    if: github.repository == 'dotcms/core'
+    environment: trunk
+    steps:
+      - name: Context
+        run: |
+          echo "Generationg a Full Starter..."
```
Comment (nollymar) [inline .github/workflows/publish-starter.yml:35]: there is a typo here: Generati**o**ng

File: .github/workflows/publish-starter.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        uses: slackapi/slack-github-action@v1.26.0
+        with:
+          channel-id: "log-starter"
+          payload: |
+            {
+              "blocks": [              
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
```
Comment (nollymar) [inline .github/workflows/publish-starter.yml:147]: can we format the artifactory url as a link instead of a code segment?

Comment (dcolina) [reply .github/workflows/publish-starter.yml:147]: Yes of course.

File: .github/workflows/publish-starter.yml (hunk 1 above)
Comment (dcolina) [reply .github/workflows/publish-starter.yml:35]: Thanks.

**Review by nollymar — APPROVED** [review]

**Review by spbolton — CHANGES_REQUESTED** [review]
If we could make DOT_STARTER_URL a variable rather than a secret,  I was thinking the same for the Sonarqube url.   The problem is that 1, it really does not need to be a secret, but 2.  It causes the url to be obfuscated in the logs, In the case of sonarqube it reports the url to the report that is generated but due to the url being a secret it removes that part from the log making it more difficult to just copy the path

**Comment by dcolina** [comment]
> If we could make DOT_STARTER_URL a variable rather than a secret, I was thinking the same for the Sonarqube url. The problem is that 1, it really does not need to be a secret, but 2. It causes the url to be obfuscated in the logs, In the case of sonarqube it reports the url to the report that is generated but due to the url being a secret it removes that part from the log making it more difficult to just copy the path

Done.

**Review by spbolton — APPROVED** [review]

**Pull request merged**